lib/components/form/batch-settings.tsx

```diff
@@ -56,35 +72,84 @@ export function setModeButtonEnabled(enabledKeys: string[]) {
 function BatchSettings({
   activeSearch,
   currentQuery,
+  departArrive,
   enabledModeButtons,
   fillModeIcons,
+  homeTimezone,
   modeButtonOptions,
   onPlanTripClick,
   openAdvancedSettings,
+  routingQuery,
   setQueryParam,
-  spacedOutModeSelector
+  sort,
+  syncSortWithDepartArrive,
+  updateItineraryFilter
 }: Props) {
   const intl = useIntl()
 
-  // Whether the date/time selector is open
-  const [dateTimeOpen, setDateTimeOpen] = useState(false)
-
   // @ts-expect-error Context not typed
   const { ModeIcon } = useContext(ComponentContext)
 
   const processedModeButtons = modeButtonOptions.map(
     pipe(addModeButtonIcon(ModeIcon), setModeButtonEnabled(enabledModeButtons))
   )
 
+  const _planTrip = useCallback(() => {
+    alertUserTripPlan(intl, currentQuery, onPlanTripClick, routingQuery)
+  }, [currentQuery, intl, onPlanTripClick, routingQuery])
+
   const baseColor = getBaseColor()
 
   const accentColor = getDarkenedBaseColor()
 
+  const onQueryParamChange = useCallback(
+    (params) => {
+      setQueryParamMiddleware(
+        syncSortWithDepartArrive,
+        updateItineraryFilter,
+        params,
+        setQueryParam,
+        sort
+      )
+    },
+    [syncSortWithDepartArrive, updateItineraryFilter, setQueryParam, sort]
+  )
+
+  const dtSelectorOpen = departArrive !== 'NOW'
+  const dtTimeInput = document.querySelector(
+    ".date-time-selector input[type='time']"
+  )
```

Comment:
Move this to inside the `useEffect` callback. (Remove from the `useEffect` references.)